util/src/test/java/io/kubernetes/client/informer/cache/ReflectorRunnableTest.java

```diff
@@ -215,4 +217,42 @@ public V1PodList list(CallGeneratorParams params) throws ApiException {
       reflectorRunnable.stop();
     }
   }
+
+  @Test
+  public void testReflectorListShouldHandleExpiredResourceVersion() throws ApiException {
+    String expectedResourceVersion = "100";
+    when(listerWatcher.list(any()))
+        .thenReturn(
+            new V1PodList().metadata(new V1ListMeta().resourceVersion(expectedResourceVersion)));
+    // constantly failing watches will make the reflector run only one time
+    when(listerWatcher.watch(any())).thenThrow(new ApiException(HttpURLConnection.HTTP_GONE, ""));
```

Comment:
What about the scenario wherein this API returns responseCode 200 but the body has an event that has status.code=410 ?
https://github.com/kubernetes-client/java/pull/1611/files#diff-31beacc81d066c2ab68de67a3587fee7bd7ef10d9ba42dc0a180b621751c6bffR226